web/components/UserMessage.jsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import EventBlueprint from './EventBlueprint.jsx';
+import { GREY } from '../utils/colors';
+import './chat.css';
+
+/**
+ * @param {object} props
+ * @returns {JSX} user message container.
+ */
+export default function UserMessage(props) {
+  const { event } = props;
+  const name = event.context.user.fullName ? event.context.user.fullName : event.context.user.name;
+  const message = `${((event.context) && (event.context.user)) ?
+    name : 'User'} has  ${((event.context) && (event.context.isActive)) ?
```

Comment:
Will it break if `context` doesn't exist in event?
Is ES6 supported? How about using optional here? `event?.context?.user?.fullName`

Reply:
@LongweiDeng I'm not sure about this, it's actually just a copy/paste from what was there before (https://github.com/salesforce/refocus-bot-timeline/pull/25/files#diff-6d403394a1a35109e563f4c7b6848580d7185465d286d6c200c6ecbf4dc9caf3L22). I think all of these are showing as diffs because I renamed the file to `.jsx`.

I can take a look at using optional here if you'd like but I think the scope of this story is growing quite a lot. I actually didn't know about optional chaining in JavaScript so that's very nice!

Reply:
@kriscfoster  No worries. There are definitely more can be refactored. But just keep the scope to the requirements of your story.